tests/tools/s3/utils.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+import json
+import logging
+import os
+import re
+import sys
+
+import boto3
+import botocore.client
+
+
+def resource_has_tags(resource, tags={}):
+    """Checks if a resource defines all `tags`."""
+    return all(
+        tag_key in resource.keys() and resource[tag_key] == tag_value
+        for (tag_key, tag_value) in tags.items()
+    )
+
+
+class S3ResourceFetcher:
+    """A class for fetching vmm-reference test resources from S3."""
+
+    def __init__(
+            self,
+            resource_bucket,
+            resource_manifest_path,
+            download_location=None,
+    ):
+        """Initializes the S3 client, manifest of test resources and S3 bucket name."""
+        with open(resource_manifest_path) as json_file:
+            self._resource_manifest = json.load(json_file)
+        self._resource_bucket = resource_bucket
+        self._s3 = boto3.client(
+            's3',
+            config=botocore.client.Config(signature_version=botocore.UNSIGNED)
+        )
+        if download_location is None:
+            path = os.path.join(
+                os.path.dirname(__file__),
+                "../../../resources")
+            self._default_download_location = os.path.abspath(path)
+        else:
+            self._default_download_location = download_location
+        self.log = logging.getLogger(__name__)
```

Comment:
Generally prefer a module level logger, So that all the functions can use it. If indeed you need a class level logger, better to specify `self.__class__.__name__` in this case would evaluate to the `module_name` which is `s3.utils`.